app/lib/payment_gateways/stripe_crypt.rb

```diff
@@ -36,21 +32,41 @@ def customer
       @customer ||= find_or_create_customer
     end
 
+    def update_billing_address(billing_address)
+      begin
+        Stripe.api_key = api_key  # Set actual Stripe secret key
+        latest_payment_method_id = latest_payment_method_id_for_customer
+        return true unless latest_payment_method_id.present?
+        payment_method = Stripe::PaymentMethod.retrieve(latest_payment_method_id)
+        update_and_save_billing_details(payment_method, billing_address)
+      rescue Stripe::StripeError => stripe_error
+        handle_stripe_error(stripe_error)
+      ensure
+        reset_stripe_api_key
+      end
+    end
+
     private
 
     delegate :payment_detail, to: :account
 
     def find_or_create_customer
       customer_id = payment_detail.credit_card_auth_code
       return create_customer if customer_id.blank?
+      retrieve_customer(customer_id)
+    end
 
-      begin
-        customer = Stripe::Customer.retrieve(customer_id, api_key)
-        return create_customer if customer.deleted?
-        customer
-      rescue Stripe::InvalidRequestError
-        create_customer
-      end
+    def update_and_save_billing_details(payment_method, billing_address)
+      update_billing_details(payment_method, billing_address)
+      payment_method.save
+    end
+
+    def retrieve_customer(customer_id)
```

Comment:
What's the purpose of this refactor?... :thinking: 

I'd probably keep it as it is, the method is not too long, and splitting it I think makes it less readable.